src/main/java/com/redhat/jenkins/plugins/cachet/matrix/CachetMatrixAction.java

```diff
@@ -0,0 +1,151 @@
+package com.redhat.jenkins.plugins.cachet.matrix;
+
+import com.redhat.jenkins.plugins.cachet.CachetJobProperty;
+import com.redhat.jenkins.plugins.cachet.ResourceProvider;
+import hudson.matrix.Combination;
+import hudson.matrix.MatrixConfiguration;
+import hudson.matrix.MatrixProject;
+import hudson.model.Action;
+import hudson.model.Item;
+import hudson.model.JobProperty;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.StaplerProxy;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class CachetMatrixAction implements Action, StaplerProxy {
+    private MatrixProject project;
+
+    public CachetMatrixAction(MatrixProject project) {
+        this.project = project;
+    }
+
+    @Override
+    public String getIconFileName() {
+        return this.project.hasPermission(Item.CONFIGURE) ?
+                "plugin/cachet-gating/images/cachet-icon.png" : null;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Cachet Configuration";
+    }
+
+    @Override
+    public String getUrlName() {
+        return "cachet-matrix";
+    }
+
+    @Override
+    public Object getTarget() {
+        this.project.checkPermission(Item.CONFIGURE);
+        return this;
+    }
+
+    // For UI purposes - does the user checked this combination?
+    public boolean isChecked(String combination) {
+        JobProperty jobProperty = getCachetJobPropertyFromCombination(combination);
+        if (jobProperty != null) return ((CachetJobProperty) jobProperty).getRequiredResources();
+        return false;
+    }
+
+    // For UI purposes - does the user selected this resource?
+    public boolean isSelected(String combination, String resource) {
+        return getCachetResources(combination).contains(resource);
+    }
+
+    // For UI purposes - get all the resources
+    public List<String> getResourceNames() {
+        return ResourceProvider.SINGLETON.getResourceNames();
+    }
+
+    /**
+     * @return all the possible combinations
+     */
+    public Iterable<Combination> getAxis() {
+        return project.getAxes().list();
+    }
+
+    /**
+     * Help method to avoid code duplication
+     * @param combination
+     * @return CachetJobProperty or null if not exist
+     */
+    private JobProperty getCachetJobPropertyFromCombination(String combination){
```

Comment:
If would be nicer to client for code to return `CachetJobProperty` and effectively move the cast from many places to just one - here.

Reply:
Thanks, done :)